sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -491,14 +514,27 @@ private BundleProcessor createBundleProcessor(
       }
     }
 
-    // Instantiate a State API call handler depending on whether a State ApiServiceDescriptor
-    // was specified.
-    HandleStateCallsForBundle beamFnStateClient =
-        bundleDescriptor.hasStateApiServiceDescriptor()
-            ? new BlockTillStateCallsFinish(
-                beamFnStateGrpcClientCache.forApiServiceDescriptor(
-                    bundleDescriptor.getStateApiServiceDescriptor()))
-            : new FailAllStateCallsForBundle(processBundleRequest);
+    // Instantiate a State API call handler depending on whether a State ApiServiceDescriptor was
+    // specified.
+    HandleStateCallsForBundle beamFnStateClient;
+    if (bundleDescriptor.hasStateApiServiceDescriptor()) {
```

Comment:
@amaliujia @anthonyqzhu 
I was under the impression that we would place the cache behind an experiment instead of opting everyone into its usage.

Reply:
We merged this assuming it was okay after seeing the passing tests / TAP presubmit passing as well. 

I can introduce an experiment in front of this block today if necessary

Reply:
Got it let's file a patch to hide the cache by an experiment.